aptos-move/framework/src/natives/aggregator_natives/context.rs

```diff
@@ -63,13 +64,58 @@ impl<'a> NativeAggregatorContext<'a> {
         let NativeAggregatorContext {
             aggregator_data, ..
         } = self;
-        let (_, destroyed_aggregators, aggregators, _snapshots) =
+        let (_, destroyed_aggregators, aggregators, snapshots) =
             aggregator_data.into_inner().into();
 
         let mut aggregator_v1_changes = HashMap::new();
         let mut aggregator_v2_changes = HashMap::new();
 
-        // First, process all writes and deltas.
+        // First process all snapshots (they need access to aggregators)
+        for (id, snapshot) in snapshots {
+            let state = snapshot.into();
+            let change = match state {
+                AggregatorSnapshotState::Create {
+                    value: SnapshotValue::Integer(value),
+                } => Some(AggregatorChange::Create(AggregatorValue::Snapshot(value))),
+                AggregatorSnapshotState::Create {
+                    value: SnapshotValue::String(value),
+                } => Some(AggregatorChange::Create(AggregatorValue::Derived(value))),
+                AggregatorSnapshotState::Delta {
+                    base_aggregator,
+                    delta,
+                } => {
+                    let delta_op = aggregators.get(&AggregatorVersionedID::V2(base_aggregator)).map_or_else(
+                        || DeltaOp::new(delta, u128::MAX, DeltaHistory::new()),
```

Comment:
why?

Reply:
In the context (AggregatorSnapshotState), we don't keep history/speculative value in the snapshot, only in the aggregator - so that it is not duplicated - and it is guaranteed to be validated fully (more context here - https://aptos-org.slack.com/archives/C057VPCD7QX/p1694555982793999 ) 

In the WriteSet, in order not to need to go to the aggregator to check history (but not use it's delta), we are extracting the DeltaOp and putting it there. 

DeltaOp::new(delta, u128::MAX, DeltaHistory::new()) is basically an empty delta op where checking history always succeeds.